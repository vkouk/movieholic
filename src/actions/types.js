export const AUTH_USER = 'auth_user';
export const UNAUTH_USER = 'unauth_user';
export const UPDATE_USER = 'update_user';
export const AUTH_ERROR = 'auth_error';

export const FETCH_MOVIES = 'fetch_movies';
export const GET_MOVIE = 'get_movie';

export const FETCH_SERIES = 'fetch_series';
export const GET_SERIE = 'get_serie';

export const ADD_TO_CART = 'add_to_cart';
export const CALCULATE_CART_TOTAL = 'calculate_cart_total';
export const GET_CART = 'get_cart';
export const ORDER_CART_ITEMS = 'order_cart_items';
export const RETURN_ORDER = 'return_order_items';
export const GET_ORDER = 'get_order';
export const GET_ORDERS = 'get_orders';
export const GET_USER_ORDERS = 'get_user_orders';
export const REMOVE_MOVIE_CART_ITEM = 'remove_movie_cart_item';
export const REMOVE_SERIE_CART_ITEM = 'remove_serie_cart_item';
export const ORDER_ERROR = 'order_error';

export const GET_USER_RECOMMENDATIONS = 'get_user_recommendations';
export const CLEAR_RECOMMENDATIONS = 'clear_recommendations';

export const FETCH_DASHBOARD_ORDERS = 'fetch_dashboard_orders';
export const FETCH_DASHBOARD_MEMBERS = 'fetch_dashboard_members';
export const FETCH_DASHBOARD_MOST_RENTED = 'fetch_dashboard_most_rented';

export const FETCH_ERROR = 'fetch_error';