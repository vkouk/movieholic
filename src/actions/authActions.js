import axios from "axios";
import { Config } from "../utils/Config";
import { AUTH_USER, UNAUTH_USER, AUTH_ERROR, UPDATE_USER } from "./types";
import setAuthToken from "../utils/AuthToken";

export const loginUser = ({ email, password }) => async dispatch => {
  await axios
    .post(`${Config.API_URL}/user/login`, { email, password })
    .then(({ data }) => {
      localStorage.setItem("token", data.token);
      setAuthToken(data.token);
      dispatch({ type: AUTH_USER, payload: data.user });
    })
    .catch(error =>
      dispatch({ type: AUTH_ERROR, payload: error.response.data })
    );
};

export const registerUser = (userData, history) => async dispatch => {
  await axios
    .post(`${Config.API_URL}/user/register`, userData)
    .then(() => history.push("/"))
    .catch(error =>
      dispatch({ type: AUTH_ERROR, payload: error.response.data })
    );
};

export const logoutUser = () => dispatch => {
  localStorage.removeItem("token");
  setAuthToken(false);
  dispatch({ type: UNAUTH_USER, payload: {} });
};

export const getProfile = id => async dispatch => {
  await axios
    .get(`${Config.API_URL}/user/profile/${id}`)
    .then(({ data }) => dispatch({ type: AUTH_USER, payload: data }))
    .catch(error =>
      dispatch({ type: AUTH_ERROR, payload: error.response.data })
    );
};

export const updateProfile = (profileValues, id) => async dispatch => {
  await axios
    .post(`${Config.API_URL}/user/profile/${id}`, profileValues)
    .then(({ data }) => dispatch({ type: UPDATE_USER, payload: data }))
    .catch(error =>
      dispatch({ type: AUTH_ERROR, payload: error.response.data })
    );
};

export const addUserBalance = values => async dispatch => {
  await axios
    .post(`${Config.API_URL}/user/addbalance`, values)
    .then(({ data }) => dispatch({ type: UPDATE_USER, payload: data }))
    .catch(error =>
      dispatch({ type: AUTH_ERROR, payload: error.response.data })
    );
};
