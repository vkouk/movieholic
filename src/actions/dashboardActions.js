import { Config } from '../utils/Config';
import axios from 'axios';
import { FETCH_DASHBOARD, FETCH_ERROR } from './types';

export const getLatestOrders = () => async dispatch => {
    await axios.get(`${Config.API_URL}/dashboard/orders`)
        .then(({ data }) => dispatch({ type: FETCH_DASHBOARD, order_payload: data }))
        .catch(error => dispatch({ type: FETCH_ERROR, payload: error.response.data }));
};

export const getLatestMembers = () => async dispatch => {
    await axios.get(`${Config.API_URL}/dashboard/members`)
        .then(({ data }) => dispatch({ type: FETCH_DASHBOARD, member_payload: data }))
        .catch(error => dispatch({ type: FETCH_ERROR, payload: error.response.data }));
};

export const getMostRented = () => async dispatch => {
    await axios.get(`${Config.API_URL}/dashboard/most-rented`)
        .then(({ data }) => dispatch({ type: FETCH_DASHBOARD, rented_payload: data }))
        .catch(error => dispatch({ type: FETCH_ERROR, payload: error.response.data }));
};