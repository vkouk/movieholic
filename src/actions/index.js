export * from './authActions';
export * from './rentActions';
export * from './movieActions';
export * from './serieActions';
export * from './recommendationActions';
export * from './dashboardActions';