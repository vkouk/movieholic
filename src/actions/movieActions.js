import axios from 'axios';
import { FETCH_MOVIE, FETCH_ERROR } from './types';

let API_URL = 'http://movieholic-api.herokuapp.com/api';

export const fetchMovies = () => async dispatch => {
    await axios.get(`${API_URL}/movie`)
        .then(({ data }) => dispatch({ type: FETCH_MOVIE, payload: data }))
        .catch(error => dispatch({ type: FETCH_ERROR, payload: error.response.data }));
};