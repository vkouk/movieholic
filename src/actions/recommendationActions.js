import axios from "axios";
import { Config } from "../utils/Config";
import {
  GET_USER_RECOMMENDATIONS,
  CLEAR_RECOMMENDATIONS,
  FETCH_ERROR
} from "./types";

export const getUserRecommendations = userId => async dispatch => {
  await axios
    .post(`${Config.API_URL}/recommendations`, userId)
    .then(({ data }) =>
      dispatch({ type: GET_USER_RECOMMENDATIONS, payload: data })
    )
    .catch(error =>
      dispatch({ type: FETCH_ERROR, payload: error.response.data })
    );
};

export const clearRecommendations = () => async dispatch => {
  dispatch({ type: CLEAR_RECOMMENDATIONS });
};
