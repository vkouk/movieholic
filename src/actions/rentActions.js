import axios from 'axios';
import { Config } from '../utils/Config';
import {
    ORDER_CART_ITEMS, ADD_TO_CART, GET_CART, CALCULATE_CART_TOTAL, ORDER_ERROR, GET_ORDER, GET_ORDERS, REMOVE_MOVIE_CART_ITEM, REMOVE_SERIE_CART_ITEM, GET_USER_ORDERS
} from './types';

export const storeRent = (rentValues, history) => async dispatch => {
    await axios.post(`${Config.API_URL}/rent`, rentValues)
        .then(() => {
            dispatch({ type: ORDER_CART_ITEMS });
            history.push('/');
        })
        .catch(error => dispatch({ type: ORDER_ERROR, payload: error.response.data }));
};

export const getRents = () => async dispatch => {
    await axios.get(`${Config.API_URL}/rent`)
        .then(({ data }) => dispatch({ type: GET_ORDERS, payload: data }))
        .catch(error => dispatch({ type: ORDER_ERROR, payload: error.response.data }));
};

export const getRent = rentId => async dispatch => {
    await axios.get(`${Config.API_URL}/rent/${rentId}`)
        .then(({ data }) => dispatch({ type: GET_ORDER, payload: data }))
        .catch(error => dispatch({ type: ORDER_ERROR, payload: error.response.data }));
};

export const getUserRents = userId => async dispatch => {
    await axios.get(`${Config.API_URL}/rent/user/${userId}`)
        .then(({ data }) => dispatch({ type: GET_USER_ORDERS, payload: data }))
        .catch(error => dispatch({ type: ORDER_ERROR, payload: error.response.data }));
};

export const addToCart = cartValues => async dispatch => {
    dispatch({ type: ADD_TO_CART, payload: cartValues });
};

export const getCart = () => async dispatch => {
    dispatch({ type: GET_CART });
};

export const cartTotalAmount = total => async dispatch => {
    dispatch({ type: CALCULATE_CART_TOTAL, payload: total });
};

export const removeMovieFromCart = movieId => async dispatch => {
    dispatch({ type: REMOVE_MOVIE_CART_ITEM, payload: movieId });
};

export const removeSerieFromCart = serieId => async dispatch => {
    dispatch({ type: REMOVE_SERIE_CART_ITEM, payload: serieId });
};