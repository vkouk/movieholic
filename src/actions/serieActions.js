import axios from 'axios';
import { Config } from '../utils/Config';
import { FETCH_SERIES, GET_SERIE, FETCH_ERROR } from './types';

export const fetchSeries = () => async dispatch => {
    await axios.get(`${Config.API_URL}/serie`)
        .then(({ data }) => dispatch({ type: FETCH_SERIES, payload: data }))
        .catch(error => dispatch({ type: FETCH_ERROR, payload: error.response.data }));
};

export const getSerie = title => async dispatch => {
    await axios.get(`${Config.API_URL}/serie/${title}`)
        .then(({ data }) => dispatch({ type: GET_SERIE, payload: data }))
        .catch(error => dispatch({ type: FETCH_ERROR, payload: error.response.data }));
};