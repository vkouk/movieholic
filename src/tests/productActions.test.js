import configureMockStore from "redux-mock-store";
import expect from "expect";
import thunk from "redux-thunk";
import * as actions from "../actions";
import * as types from "../actions/types";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe("movie/serie actions", () => {
  const store = mockStore();

  beforeEach(() => {
    store.clearActions();
  });

  describe("fetch movies/series", () => {
    it("should fetch data", () => {
      const moviesData = {
        type: types.FETCH_MOVIES,
        payload: {
          id: 1,
          title: "batman",
          stock: 1,
          genre: "Action",
          released: "25/12/2018"
        }
      };
      const seriesData = {
        type: types.FETCH_SERIES,
        payload: {
          id: 1,
          title: "daredevil",
          stock: 1,
          genre: "Action",
          released: "25/12/2018"
        }
      };

      const expectedActions = [
        {
          type: types.FETCH_MOVIES,
          payload: moviesData.payload
        },
        {
          type: types.FETCH_SERIES,
          payload: seriesData.payload
        }
      ];

      store.dispatch(actions.fetchMovies()).then(() => {
        expect(store.getActions()).toEqual(expectedActions);
      });

      store.dispatch(actions.fetchSeries()).then(() => {
        expect(store.getActions()).toEqual(expectedActions);
      });
    });

    it("should correct fetch data", () => {
      store.dispatch(actions.fetchMovies());
      store.dispatch(actions.fetchSeries());
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe("fetch movie/serie by title", () => {
    it("should fetch data by title", () => {
      const moviesData = {
        type: types.GET_MOVIE,
        payload: { title: "batman" }
      };
      const seriesData = {
        type: types.GET_SERIE,
        payload: { title: "daredevil" }
      };

      const expectedActions = [
        {
          type: types.GET_MOVIE,
          payload: moviesData.payload
        },
        {
          type: types.GET_SERIE,
          payload: seriesData.payload
        }
      ];

      store.dispatch(actions.fetchMovies()).then(() => {
        expect(store.getActions()).toEqual(expectedActions);
      });

      store.dispatch(actions.fetchSeries()).then(() => {
        expect(store.getActions()).toEqual(expectedActions);
      });
    });

    it("should correct fetch data by title", () => {
      store.dispatch(actions.getMovie("batman"));
      store.dispatch(actions.getSerie("daredevil"));
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
