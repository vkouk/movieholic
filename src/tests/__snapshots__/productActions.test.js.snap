// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movie/serie actions fetch movie/serie by title should correct fetch data by title 1`] = `Array []`;

exports[`movie/serie actions fetch movies/series should correct fetch data 1`] = `Array []`;
